### PR TITLE
[onepunchman] added website support

### PR DESCRIPTION
**Pull request by XenonR**
Added website support for https://ww3.one-punchman.com/

### CODE REVIEW

File: src/web/mjs/connectors/OnePunchMan.mjs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        super.id = 'onepunchman';
+        super.label = 'One-Punch Man';
+        this.tags = [ 'manga', 'english', 'high-quality' ];
+        this.url = 'https://ww3.one-punchman.com/';
+    }
+
+    async _getMangas() {
+        return [
+            {
+                id: 'onepunchman',
```
Comment (27136773) [inline src/web/mjs/connectors/OnePunchMan.mjs]: I suggest to use a manga identifier (even if there are no other mangas) to be compliant with the other connectors, e.g.
`id: this.url`
`id: '/'`

File: src/web/mjs/connectors/OnePunchMan.mjs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                title: this.label
+            }
+        ];
+    }
+
+    async _getChapters() {
+        let request = new Request(this.url, this.requestOptions);
```
Comment (27136773) [inline src/web/mjs/connectors/OnePunchMan.mjs]: Same here, include the manga identifier, e.g.
`new Request(new URL(manga.id, this.url), this.requestOptions)`

File: src/web/mjs/connectors/OnePunchMan.mjs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        let data = await this.fetchDOM(request, '#Chapters_List a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: element.text.split(", ")[1],
```
Comment (27136773) [inline src/web/mjs/connectors/OnePunchMan.mjs]: May prefer single quotes instead of double quotes

File: src/web/mjs/connectors/OnePunchMan.mjs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            }
+        ];
+    }
+
+    async _getChapters() {
+        let request = new Request(this.url, this.requestOptions);
+        let data = await this.fetchDOM(request, '#Chapters_List a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
```
Comment (27136773) [inline src/web/mjs/connectors/OnePunchMan.mjs]: May use `this.url` instead of `request.url`

File: src/web/mjs/connectors/OnePunchMan.mjs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    async _getPages(chapter) {
+        let request = new Request(this.url + chapter.id, this.requestOptions);
+        let data = await this.fetchDOM(request, '.entry-content a');
+        return data.map(element => this.getAbsolutePath(element, request.url));
+    }
+
+    async _getMangaFromURI(uri) {
+        let id = this.getRootRelativeOrAbsoluteLink(uri.href, this.url);
```
Comment (27136773) [inline src/web/mjs/connectors/OnePunchMan.mjs]: May use same ID as applied in `_getMangas`, e.g.
`id: this.url`
`id: '/'`

File: src/web/mjs/connectors/OnePunchMan.mjs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        let title = this.label;
+        return new Manga(this, id, title);
```
Comment (27136773) [inline src/web/mjs/connectors/OnePunchMan.mjs]: May skip local variables and return directly, e.g.
`return new Manga(this, this.url, this.label);`

**Comment by XenonR** [comment]
Sorry for the mess

**Pull request closed without merging**